ui/component/claimPreview/view.jsx

```diff
@@ -292,6 +297,14 @@ const ClaimPreview = forwardRef<any, {}>((props: Props, ref: any) => {
     }
   }
 
+  function removeFromHistory(e, uri) {
+    e.stopPropagation();
+    if (watchHistory.find((entry) => entry === uri)) {
+      watchHistory.splice(watchHistory.indexOf(uri), 1);
```

Comment:
This whole chunk should go away when we move to Redux, but wanted to note that React states must be immutable to avoid unwanted side-effects.  Just like in Redux, make a copy through `slice()` or some other means.